AK/FixedArray.h

```diff
@@ -36,6 +36,42 @@ class FixedArray {
         return MUST(try_create(size));
     }
 
+    // NOTE:
+    // Even though it may look like there will be a template instantiation of these functions for every size,
+    // the compiler will inline these anyway and therefore not generate any duplicate code.
+
+    template<size_t N>
+    static ErrorOr<FixedArray<T>> try_create(T(&&array)[N])
+    {
+        if (N == 0)
+            return FixedArray<T>();
+        T* elements = static_cast<T*>(kmalloc_array(N, sizeof(T)));
+        if (!elements)
+            return Error::from_errno(ENOMEM);
+        for (size_t i = 0; i < N; ++i)
+            new (&elements[i]) T(move(array[i]));
+        return FixedArray<T>(N, elements);
+    }
+
+    template<size_t N>
+    static FixedArray<T> must_create_but_fixme_should_propagate_errors(T(&&array)[N])
+    {
+        return MUST(try_create(move(array)));
+    }
```

Comment:
Where is this used? We usually don't add dead code.

Reply:
This is part of #11805 where we already had a small discussion about it. To quote my argument from there:
> The reason I think these APIs are useful is that it extremely lowers the barrier to convert a user to `FixedArray` from some other class like `Vector` that's not OOM-aware. If you had previously created the latter in a constructor, you can now simply use the `must_foo_but_fixme_should_propagate_errors()` to create the former. But if we force every new user to use error propagation, they will need to restructure a lot more code in order to make the switch.